sdk/communication/Azure.Communication.JobRouter/src/RouterClient.cs

```diff
@@ -0,0 +1,1345 @@
+        /// <param name="endpoint">The URI of the Azure Communication Services resource.</param>
+        /// <param name="keyCredential">The <see cref="AzureKeyCredential"/> used to authenticate requests.</param>
+        /// <param name="options">Client option exposing <see cref="ClientOptions.Diagnostics"/>, <see cref="ClientOptions.Retry"/>, <see cref="ClientOptions.Transport"/>, etc.</param>
+        public RouterClient(Uri endpoint, AzureKeyCredential keyCredential, RouterClientOptions options = default)
+            : this(
+                Argument.CheckNotNull(endpoint, nameof(endpoint)).AbsoluteUri,
+                Argument.CheckNotNull(keyCredential, nameof(keyCredential)),
+                options ?? new RouterClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a new instance of <see cref="RouterClient"/>.</summary>
+        /// <param name="endpoint">The URI of the Azure Communication Services resource.</param>
+        /// <param name="tokenCredential">The TokenCredential used to authenticate requests, such as DefaultAzureCredential.</param>
+        /// <param name="options">Client option exposing <see cref="ClientOptions.Diagnostics"/>, <see cref="ClientOptions.Retry"/>, <see cref="ClientOptions.Transport"/>, etc.</param>
+        public RouterClient(Uri endpoint, TokenCredential tokenCredential, RouterClientOptions options = default)
+            : this(
+                Argument.CheckNotNull(endpoint, nameof(endpoint)).AbsoluteUri,
+                Argument.CheckNotNull(tokenCredential, nameof(tokenCredential)),
+                options ?? new RouterClientOptions())
+        {
+        }
+
+        #endregion
+
+        #region private constructors
+
+        private RouterClient(ConnectionString connectionString, RouterClientOptions options)
+            : this(connectionString.GetRequired("endpoint"), options.BuildHttpPipeline(connectionString), options)
+        {
+        }
+
+        private RouterClient(string endpoint, TokenCredential tokenCredential, RouterClientOptions options)
+            : this(endpoint, options.BuildHttpPipeline(tokenCredential), options)
+        {
+        }
+
+        private RouterClient(string endpoint, AzureKeyCredential keyCredential, RouterClientOptions options)
+            : this(endpoint, options.BuildHttpPipeline(keyCredential), options)
+        {
+        }
+
+        private RouterClient(string endpoint, HttpPipeline httpPipeline, RouterClientOptions options)
+        {
+            _clientDiagnostics = new ClientDiagnostics(options);
+            RestClient = new JobRouterRestClient(_clientDiagnostics, httpPipeline, endpoint, options.ApiVersion);
+        }
+
+        #endregion
+
+        /// <summary>Initializes a new instance of <see cref="RouterClient"/> for mocking.</summary>
+        protected RouterClient()
+        {
+            _clientDiagnostics = null;
+            RestClient = null;
+        }
```

Comment:
Can we move this at the top with constructors section?

Reply:
Done